### PR TITLE
User Story AB#1239723: [Infra] [Scripting] Add Custom Components Sample to minecraft-scripting-samples, and add a tutorial to go along with it

### PR DESCRIPTION
**Pull request by optic5471ms**
# Description

Added a sample pack for item and block custom components. This pack adds a strawberry crop, food item, and a farmland watering sprayer item.

### CODE REVIEW

**Comment by optic5471ms** [comment]
> @optic5471ms please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

**Review by SBLMikeDemone — COMMENTED** [review]
Looks really good! I'll pull it down and mess around with it just to make sure it all works on my end

File: custom-components-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ms.date: 05/05/2022
+languages:
+  - typescript
+products:
+  - minecraft
+---
+
+# Minecraft How-to Gallery
+
+This sample contains many code snippets that show how you can manipulate basic Minecraft behaviors.
```
[the remark this answers is not in the document]
Comment (SBLMikeDemone) [reply custom-components-sample/README.md:15]: Oh actually it looks like the top level readme is where the samples are explained

File: custom-components-sample/behavior_packs/custom_components_sample/manifest.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        0,
+        0
+      ],
+      "entry": "scripts/main.js"
+    }
+  ],
+  "dependencies": [
+    {
+      "module_name": "@minecraft/server",
+      "version": "1.13.0-beta"
```
[the remark this answers is not in the document]
Comment (optic5471ms) [reply custom-components-sample/behavior_packs/custom_components_sample/manifest.json:35]: [Task 1248195](https://dev-mc.visualstudio.com/Minecraft/_workitems/edit/1248195): Update custom component sample to use 1.12 when custom components are stabilized

This task will stay on me until 1.12 is released and then we can update this pack

File: custom-components-sample/behavior_packs/custom_components_sample/items/sprayer_full.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+			"minecraft:icon": "spray_can_full",
+			"minecraft:liquid_clipped": true,
+      "minecraft:custom_components": [
+        "starter:spray_water"
+      ],
+			"minecraft:use_modifiers": {
+				"use_duration": 1,
+				"movement_modifier": 0.35
+			},
+			"minecraft:use_animation": "drink"
```
[the remark this answers is not in the document]
Comment (optic5471ms) [reply custom-components-sample/behavior_packs/custom_components_sample/items/sprayer_full.json:17]: Yup! Thats how "charge" it to spray out the water

**Comment by mammerla** [comment]
Could you remove the -sample from the root folder name (and maybe in the name in various package.json type things?)

**Pull request merged**